src/DynamoUtilities/DynamoFeatureFlagsManager.cs

```diff
@@ -91,6 +105,13 @@ internal void CacheAllFlags()
         /// <param name="defaultval">Currently the flag and default val MUST be a bool or string.</param>
         /// <returns></returns>
         internal T CheckFeatureFlag<T>(string featureFlagKey, T defaultval)
+        {
+            // with testmode = true, the call goes through an interface so that we can intercept it with Mock
+            // with testmode = false, the call simply goes to the CheckFeatureFlagInternal
+            return testmode ? flags.CheckFeatureFlag(this, featureFlagKey, defaultval) : CheckFeatureFlagInternal(featureFlagKey, defaultval);
```

Comment:
just a performance thing I guess

Reply:
Isn't `flags.CheckFeatureFlag()` also calling `CheckFeatureFlagInternal()`? How is the `testmode `case any different from the normal case?